tests/scripts/icmp_responder.py

```diff
@@ -0,0 +1,62 @@
+import argparse
+
+from scapy.all import conf, Ether, ICMP, IP
+from scapy.arch import get_if_hwaddr
+from scapy.data import ETH_P_IP
+from select import select
+
+
+def respond_to_icmp_request(socket, request):
+    """Respond to ICMP request."""
+    reply = request.copy()
+    reply[ICMP].type = 0
+    # Force re-generation of the checksum
+    reply[ICMP].chksum = None
+    reply[IP].src, reply[IP].dst = request[IP].dst, request[IP].src
+    reply[IP].chksum = None
+    reply[Ether].src, reply[Ether].dst = request[Ether].dst, request[Ether].src
```

Comment:
Does this handle icmp packets coming with Vlan tag?

Reply:
If the packet comes with vlan tag, the reply packet will have the same vlan tag, is this expected?

Reply:
yes, thats expected

Reply:
Thanks, mark as resolved.